src/main/java/org/minions/devfund/abel/WhackAMole.java

```diff
@@ -0,0 +1,182 @@
+        if (isValidPosition(row, col, MOLE)) {
+            this.molesLeft--;
+            this.score++;
+            this.moleGrid[row][col] = MOLE_WHACKED;
+        }
+    }
+
+    /**
+     * Verifies if the position contains an specific char.
+     *
+     * @param row    row position to check.
+     * @param col    col position to check.
+     * @param output char value to verify.
+     * @return if the position is valid.
+     */
+    private boolean isValidPosition(int row, int col, char output) {
+        int size = this.moleGrid.length;
+        boolean checkRow = ZERO <= row && row < size;
+        boolean checkCol = ZERO <= col && col < size;
+        return checkRow && checkCol && output == this.moleGrid[row][col];
+    }
+
+    /**
+     * Print the grid without showing where the moles are.
+     * For every spot that has recorded a “whacked mole,” print out a W, or * otherwise.
+     *
+     * @return the grid completely without moles.
+     */
+    public String printGridToUser() {
+        int length = this.moleGrid.length;
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < length; i++) {
+            for (int j = 0; j < length; j++) {
+                stringBuilder.append(this.moleGrid[i][j] == MOLE ? EMPTY_MOLE : this.moleGrid[i][j]);
+            }
+            stringBuilder.append(System.lineSeparator());
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Print the grid completely. This is effectively dumping the 2d array on the screen.
+     * The places where the moles are should be printed as an ‘M’.
+     * The places where the moles have been whacked should be printed as a ‘W’.
+     * The places that don’t have a mole should be printed as *.
+     *
+     * @return the grid completely.
+     */
+    public String printGrid() {
+        int length = this.moleGrid.length;
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < length; i++) {
+            for (int j = 0; j < length; j++) {
+                stringBuilder.append(this.moleGrid[i][j]);
+            }
+            stringBuilder.append(System.lineSeparator());
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
+     * The psvm method.
+     *
+     * @param args Args.
+     */
+    public static void main(final String[] args) {
+        final int numAttempts = 50;
+        final int gridDimension = 10;
+        final int totalMoles = 10;
+        final int givingUp = -1;
+        WhackAMole whackAMole = new WhackAMole(numAttempts, gridDimension);
+        Random random = new Random();
+        do {
+            int row = random.nextInt(gridDimension);
+            int col = random.nextInt(gridDimension);
+            whackAMole.place(row, col);
+        } while (whackAMole.molesLeft < totalMoles);
+        while (whackAMole.attemptsLeft > 0) {
+            System.out.println(String.format("You have %d attempts left.", numAttempts));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)